docusaurus/docs/testing/extensions-test.md

```diff
@@ -0,0 +1,32 @@
+# Extensions Tests
+
+The Extensions framework is tested by the GitHub action `check-plugins.yaml`. This runs the script `./shell/scripts/test-plugins-build.sh`
+which runs the actual tests.
+
+The tests check the following:
+
+- Ensures that Verdaccio is installed and running locally (Verdaccio provides a local NPM registry that the tests can use to publish packages without having to publish to npm)
+- Verifies the build of the following, by building and then publishing them to the local registry
+  - Shell
+  - Yarn Creator packages
+  - Rancher components package
+- Verifies that a new app can be created with the `@rancher/app` creator and built (test case of a new extension in a new repository)
+- Verifies that an extension can be added with the `@rancher/pkg` creator to the new app and that the dashboard build passes with the new extension present
+- Verifies that the extension can be build standalone
+- Verifies that an extension can be added to the existing dashboard and that the dashboard build passes with the new extension present
+
+## Possible causs of test failures
+
+### Dependencies
+
+For the standalone case of a new app using the `@rancher/shell` package, the dependency versions can end up being different to that of the dashboard codebase. This is typically due to dependencies being top-level dependencies in the case of the dashboard codebase, but child dependencies in the case of the new app. In the later case, yarn can hoist dependency versions differently.
+
+### Imports
+
+Importing Vue components using the `~` syntax should not be used - this will build in the case of the dashboard repository, but not in the case of a standalone app.
+
+## Running Tests locally
+
+You can run the script `./shell/scripts/test-plugins-build.sh` locally in a terminal to run the test suite on a developer machine.
+
+> Note: After running the tests the `package.json` files in `shell` and `pkg/rancher-components` will be modified - ensure you reset these and do not check in changes.
```

Comment:
Can we extend the ignore list for these generated files?

```
login.json
pkg/rancher-components/package.json.bak
```

Reply:
It seems like this is the part of code which we should correct, as it checks the file but not the content:
```
 ~/.config/verdaccio/htpasswd
```

Reply:
If I try to remove the file and run again the command I get
```
[2/4] Logging in...
info npm username: admin
info npm email: admin
question npm password: 
error An unexpected error occurred: "http://localhost:4873/-/user/org.couchdb.user:admin: username is already registered".
```

Reply:
After removing manually the folders for the configuration and libraries the database still seems keeping the users and also the CLI is now broken 💀💀💀💀💀💀💀💀 

<img width="482" alt="Screenshot 2023-04-20 at 11 14 32" src="https://user-images.githubusercontent.com/5009481/233319559-66cf90f9-d237-4e2c-858b-87a6a9e7d6c7.png">

Reply:
I was able to make it work after reinstalling it multiple times 🤔

Reply:
If you have to type in the user when running the script locally, you have to logout with `yarn` pointing to the registry and not `npm` as in the script.
E.g. `yarn logout --registry=http://localhost:4873`

Reply:
Although the storage is located in `~/.local/share/verdaccio/storage/.verdaccio-db.json`, it does not seems removing the added `admin` user when changed, so the login always prints out 
```
{
  "error": "username is already registered"
}
```

Reply:
In order to update configurations, the service must be stopped with `kill` command, then restarted the script. This is the only way to also reset the database.

If you want to change the path you can either use an env var (`VERDACCIO_STORAGE_PATH `) or change the `storage:` configuration inside the `config.yaml`

Reply:
Still get 
```
npm ERR! code E401
npm ERR! Unable to authenticate, your authentication token seems to be invalid.
```